src/core/elevation/qgsabstractprofilegenerator.h

```diff
@@ -243,14 +257,39 @@ class CORE_EXPORT QgsProfileGenerationContext
      */
     void setElevationRange( const QgsDoubleRange &range ) { mElevationRange = range; }
 
+    /**
+     * Returns the scaling factor for converting pixels to physical sizes.
+     *
+     * This is usually equal to the number of pixels per millimeter.
+     *
+     * \see setScaleFactor()
+     */
+    double scaleFactor() const {return mScaleFactor;}
+
+    /**
+     * Sets the scaling factor for the render to convert pixels to physical sizes.
+     *
+     * This should usually be equal to the number of pixels per millimeter.
+     *
+     * \see scaleFactor()
+     */
+    void setScaleFactor( double factor ) {mScaleFactor = factor;}
```

Comment:
I have always disliked the "scale factor" in QgsRenderContext as it seemed very arbitrary in the rendering code that used it ("why do we need to apply scale factor here and what does it mean?"). Can it actually have other meaning than number of pixels per millimeter here? If not, I would suggest to give it a more specific name, e.g. setPixelsPerMm(...)

Reply:
I've gone with "setDpi" / "dpi" instead :+1: